src/Build/BackEnd/Components/SdkResolution/SdkResolution.cs

```diff
@@ -0,0 +1,198 @@
+        }
+
+        internal static SdkResolution Instance { get; } = new SdkResolution(new SdkResolverLoadingStrategy());
+
+        /// <summary>
+        ///     Get path on disk to the referenced SDK.
+        /// </summary>
+        /// <param name="sdk">SDK referenced by the Project.</param>
+        /// <param name="logger">Logging service.</param>
+        /// <param name="buildEventContext">Build event context for logging.</param>
+        /// <param name="importElementLocation">Location of the element within the project which referenced the SDK.</param>
+        /// <param name="solutionPath">Path to the solution if known.</param>
+        /// <returns>Path to the root of the referenced SDK.</returns>
+        internal string GetSdkPath(ReferencedSdk sdk, ILoggingService logger, BuildEventContext buildEventContext,
+            ElementLocation importElementLocation, string solutionPath)
+        {
+            ErrorUtilities.VerifyThrowInternalNull(sdk, nameof(sdk));
+            ErrorUtilities.VerifyThrowInternalNull(logger, nameof(logger));
+            ErrorUtilities.VerifyThrowInternalNull(buildEventContext, nameof(buildEventContext));
+            ErrorUtilities.VerifyThrowInternalNull(importElementLocation, nameof(importElementLocation));
+
+            if (_resolvers == null) Initialize(logger, buildEventContext, importElementLocation);
+
+            var results = new List<SdkResultImpl>();
+
+            try
+            {
+                var buildEngineLogger = new SdkLoggerImpl(logger, buildEventContext);
+                foreach (var sdkResolver in _resolvers)
+                {
+                    var context = new SdkResolverContextImpl(buildEngineLogger, importElementLocation.File, solutionPath);
+                    var resultFactory = new SdkResultFactoryImpl(sdk);
+                    try
+                    {
+                        var result = (SdkResultImpl)sdkResolver.Resolve(sdk, context, resultFactory);
+                        if (result != null && result.Success)
+                        {
+                            LogWarnings(logger, buildEventContext, importElementLocation, result);
+                            return result.Path;
+                        }
+
+                        results.Add(result);
+                    }
+                    catch (Exception e)
+                    {
+                        logger.LogFatalBuildError(buildEventContext, e, new BuildEventFileInfo(importElementLocation));
+                    }
+                }
+            }
+            catch (Exception e)
+            {
+                logger.LogFatalBuildError(buildEventContext, e, new BuildEventFileInfo(importElementLocation));
+                throw;
+            }
+
+            foreach (var result in results)
+            {
+                LogWarnings(logger, buildEventContext, importElementLocation, result);
+
+                if (result.Errors != null)
```

Comment:
I'd log errors only when:
 - there was no successful resolver
 - there was at least one successful resolver and the log verbosity is diagnostic. And in this case I'd maybe convert the errors to warnings / messages so they don't fail the build.